Modules/Core/Common/include/itkMath.h

```diff
@@ -318,25 +319,22 @@ FloatAlmostEqual(T                                        x1,
     return true;
   }
 
-#if defined(__APPLE__) && (__clang_major__ == 3) && (__clang_minor__ == 0) && defined(NDEBUG) && defined(__x86_64__)
-  Detail::FloatIEEE<T> x1f(x1);
-  Detail::FloatIEEE<T> x2f(x2);
-  double               x1fAsULP = static_cast<double>(x1f.AsULP());
-  double               x2fAsULP = static_cast<double>(x2f.AsULP());
-  double               ulps = x1fAsULP - x2fAsULP;
-  if (ulps < 0)
+  // This check for different signs is necessary for several reasons, see the blog link above.
+  // Subtracting the signed-magnitude representation of floats using twos-complement
+  // math isn't particularly meaningful, and the subtraction would produce a 33-bit
+  // result and overflow an int.
+  if (std::signbit(x1) != std::signbit(x2))
   {
-    ulps = -ulps;
+    // Check for equality to make sure +0 == -0
+    return (x1 == x2);
```

Comment:
Thanks for following my suggestion @seanm Now I'm sorry I just realize that this might still produce a warning! Like this:
> warning: comparing floating point with == or != is unsafe [-Wfloat-equal]

Reply:
Which is an oddly worded warning. There's nothing "unsafe" about such comparisons, they just aren't always what one wants/expects. But here exact comparison is what's wanted, so such a warning is a false positive. The only workaround I know is to suppress it with a pragma. I could make an `AreFloatsExactlyEqual` function to wrap it...

Reply:
@seanm  We're lucky, it's there already  😃 
 
https://github.com/InsightSoftwareConsortium/ITK/blob/9531f046a35134861386854ae491b0956f49fb60/Modules/Core/Common/include/itkMath.h#L723

Reply:
Reading the blog more carefully, and in fact we don't even need it.